pkg/metrics/metrics.go

```diff
@@ -0,0 +1,52 @@
+package metrics
+
+import (
+	"fmt"
+	logger "github.com/intel/cri-resource-manager/pkg/log"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	builtInCollectors map[string]InitCollector
+	log               = logger.NewLogger("metrics")
+)
+
+func init() {
+	builtInCollectors = make(map[string]InitCollector)
+}
+
+type InitCollector func() (prometheus.Collector, error)
+
+func RegisterCollector(name string, init InitCollector) error {
```

Comment:
nit: `init` overshadows the `init()` function here. Probably makes sense to choose a different name for the arg to avoid confusions.

Reply:
true, I'll rename

Reply:
per our chat, let's keep it